File: server/data/results/translations/lite/plain/javascript/balanced_parens.js
function is_balanced(text) {
    const stack = [];
    const pairs = {")": "(", "]": "[", "}": "{"};
    for (const ch of text) {
        if (ch === "(" || ch === "[" || ch === "{") {
            stack.push(ch);
        } else if (ch in pairs) {
            if (stack.length === 0) {
                return false;
            }
            const top = stack.pop();
            if (top !== pairs[ch]) {
                return false;
            }
        }
    }
    return stack.length === 0;
}

function main() {
    const samples = ["()", "([{}])", "(]", "((()", "{[()]}"];
    for (const sample of samples) {
        if (is_balanced(sample)) {
            console.log("true");
        } else {
            console.log("false");
        }
    }
}

main();

File: server/data/results/translations/lite/plain/java/balanced_parens.java
import java.util.ArrayDeque;
import java.util.Arrays;
import java.util.Deque;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Main {
    public static boolean isBalanced(String text) {
        Deque<Character> stack = new ArrayDeque<>();
        Map<Character, Character> pairs = new HashMap<>();
        pairs.put(')', '(');
        pairs.put(']', '[');
        pairs.put('}', '{');

        for (int i = 0; i < text.length(); i++) {
            char ch = text.charAt(i);
            if (ch == '(' || ch == '[' || ch == '{') {
                stack.push(ch);
            } else if (pairs.containsKey(ch)) {
                if (stack.isEmpty()) {
                    return false;
                }
                char top = stack.pop();
                if (top != pairs.get(ch)) {
                    return false;
                }
            }
        }
        return stack.isEmpty();
    }

    public static void main(String[] args) {
        List<String> samples = Arrays.asList("()", "([{}])", "(]", "((()", "{[()]}");
        for (String sample : samples) {
            if (isBalanced(sample)) {
                System.out.println("true");
            } else {
                System.out.println("false");
            }
        }
    }
}